streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -177,10 +199,29 @@ void openDB(final ProcessorContext context) {
             throw new ProcessorStateException(fatal);
         }
 
+        metricsRecorder.ifPresent(
+            metricsRecorder -> setUpMetrics(
+                (StreamsMetricsImpl) context.metrics(),
+                context.taskId(),
+                RecordingLevel.forName((String) configs.get(METRICS_RECORDING_LEVEL_CONFIG))
+            )
+        );
+
         openRocksDB(dbOptions, columnFamilyOptions);
         open = true;
     }
 
+    private void setUpMetrics(final StreamsMetricsImpl metrics,
+                              final TaskId taskId,
+                              final RecordingLevel recordingLevel) {
+        if (userSpecifiedOptions.statistics() == null && recordingLevel == RecordingLevel.DEBUG) {
+            final Statistics statistics = new Statistics();
+            statistics.setStatsLevel(StatsLevel.EXCEPT_DETAILED_TIMERS);
+            userSpecifiedOptions.setStatistics(statistics);
+            metricsRecorder.get().addStatistics(name, statistics, metrics, taskId);
```

Comment:
RocksDB's `Statistics` object to collect the metrics is created and passed into the recorder.